sage-impl/verify.sage

```diff
@@ -69,12 +67,28 @@ def canonical(E):
     A = min(As, key = lambda el: (int(el[0]), int(el[1])))
     return E.isomorphism_to(curve(A))
 
-def unpack_chain(A, chain):
+def unpack_chain(A, chain, deg):
     E = curve(A)
+
+    # find l-torsion basis
+    l, = deg.prime_factors()
+    pts = filter(bool, E(0).division_points(l))
```

Comment:
So this filters for all booleans?

Reply:
`filter(f, xs)` is an iterable containing all the `x` in `xs` for which `f(x)` is truthy. For elliptic-curve points in particular, `bool(P)` checks if the point is nonzero. So this iterates over all nonzero `l`-torsion points, and since `l` is prime, those are the points of order exactly `l`.

Reply:
ok; not the most obvious syntax but, Python! 🐍